app/src/main/java/com/kickstarter/viewmodels/UpdateViewModel.kt

```diff
@@ -189,46 +220,61 @@ interface UpdateViewModel {
                         )
                     }
                 }
-                .compose(bindToLifecycle())
-                .subscribe { startShareIntent.onNext(it) }
+                .subscribe {
+                    it?.let { pair ->
+                        startShareIntent.onNext(pair)
+                    }
+                }
+                .addToDisposable(disposables)
 
             goToCommentsActivity
-                .delay(2, TimeUnit.SECONDS, environment.scheduler()) // add delay to wait until activity subscribed to viewmodel
+                .delay(
```

Comment:
Probably this delay can now be removed, as the view model is created now in a different way